### DISCUSSION TITLE
Are bookmarks and highlights device agnostic?

### DISCUSSION THREAD

**Discussion by antrmn** [koreader/koreader#11207] [Q&A]
Assuming same version of KOReader on all devices, can the `bookmarks` and `highlight` be cut and copied from one device's metadata file to another file's metadata file of the same book?

PDFs have document regions (pboxes) indicated by page number + page coordinate, while ePubs do have document regions indicated by path to document elements (e.g. `/body/DocFragment[10]/body/div/p[14]/text()[1].152`). They both seem to portable data, aren't they?

I'd like to wire up a simple and primitive mechanism for keeping highlights+bookmarks in sync between my Android device and Linux laptop and I was wondering what are the main difficulties to be aware of.

Thank you in advance for your help

**Comment by Frenzie** [comment]
The docfragment stuff can be slightly different between versions (you'll note a file indicates a version it was rendered with). But if you just mean something like rsync that should work fine.

**Reply by antrmn** [reply]
You mean the `cre_dom_version` attribute in the metadata file, right? Having it with the same value would guarantee compatibility?

**Reply by poire-z** [reply]
Yes, they should.
For CRE (epubs, fb2, ...), you probably shouldn't worry, it will work (everything changes as much if you change font size and margins as if you swtich devices).
For PDF, I don't know this part enough to be 100% sure, but there's so many drawing modes that the pboxes are probably made on the original pages, so I guess it should also work fine.

Pinging @mergen3107 who may have more experience with that as he changes devices as often as I change socks :)
